lib/elements/dom-if.js

```diff
@@ -397,25 +397,34 @@ class DomIfFast extends DomIfBase {
           // the next render. Clearing `__invalidProps` here ensures
           // `_showHideChildren`'s call to `__syncHostProperties` no-ops, so
           // that we don't call `runEffects` more often than necessary.
-          this.__syncInfo = null;
+          instance.__syncInfo = null;
           this._showHideChildren();
           changedProps = Object.assign(syncInfo.changedProps, changedProps);
-          hasPaths = hasPaths || syncInfo.hasPaths;
         }
         runEffects(changedProps, hasPaths);
       } else {
         // Accumulate any values changed while `if` was false, along with the
         // runEffects method to sync them, so that we can replay them once `if`
         // becomes true
-        if (syncInfo) {
-          syncInfo.hasPaths = syncInfo.hasPaths || hasPaths;
-          Object.assign(syncInfo.changedProps, changedProps);
-        } else {
-          this.__syncInfo = {
-            runEffects,
-            changedProps: Object.assign({}, changedProps),
-            hasPaths
-          };
+        if (instance) {
+          if (!syncInfo) {
+            syncInfo = instance.__syncInfo = { runEffects, changedProps: {} };
```

Comment:
Why is it ok to drop the passed in `changedProps` here? (The next section is only for `hasPaths`.)

Reply:
They're added in the `Object.assign` below;  just a refactoring of the code.